From: Jeff Moyer <jmoyer@redhat.com>
Date: Tue, 1 Dec 2009 21:06:30 -0500
Subject: [fs] eventfd: wire up x86 arches
Message-id: <1259701600-23508-3-git-send-email-jmoyer@redhat.com>
Patchwork-id: 21612
O-Subject: [RHEL5 PATCH 02/12 v2] signal/timer/event: eventfd wire up x86 arches
Bugzilla: 493101
RH-Acked-by: Josef Bacik <josef@redhat.com>

Fixes bug 493101.

commit fdb902b1225e1668315f38e96d2f439452c03a15
Author: Davide Libenzi <davidel@xmailserver.org>
Date:   Thu May 10 22:23:20 2007 -0700

    signal/timer/event: eventfd wire up x86 arches

    This patch wires the eventfd system call to the x86 architectures.

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>

diff --git a/arch/i386/kernel/syscall_table.S b/arch/i386/kernel/syscall_table.S
index 583d25d..e86bf6b 100644
--- a/arch/i386/kernel/syscall_table.S
+++ b/arch/i386/kernel/syscall_table.S
@@ -330,5 +330,5 @@ ENTRY(sys_call_table)
 	.long sys_ni_syscall		/* 320 */ /* sys_utimensat */
 	.long sys_ni_syscall		/* sys_signalfd */
 	.long sys_ni_syscall		/* sys_timerfd_create */
-	.long sys_ni_syscall		/* sys_eventfd */
+	.long sys_eventfd		/* sys_eventfd */
 	.long sys_fallocate
diff --git a/arch/x86_64/ia32/ia32entry.S b/arch/x86_64/ia32/ia32entry.S
index 40e8168..ebfc581 100644
--- a/arch/x86_64/ia32/ia32entry.S
+++ b/arch/x86_64/ia32/ia32entry.S
@@ -736,6 +736,6 @@ ia32_sys_call_table:
 	.quad quiet_ni_syscall		/* 320 */ /* compat_sys_utimensat */
 	.quad quiet_ni_syscall		/* compat_sys_signalfd */
 	.quad quiet_ni_syscall		/* sys_timerfd_create */
-	.quad quiet_ni_syscall		/* sys_eventd */
+	.quad sys_eventfd		/* sys_eventd */
 	.quad sys32_fallocate
 ia32_syscall_end:		
diff --git a/include/asm-i386/unistd.h b/include/asm-i386/unistd.h
index 809d950..d2dba03 100644
--- a/include/asm-i386/unistd.h
+++ b/include/asm-i386/unistd.h
@@ -328,7 +328,7 @@
 /* #define __NR_utimensat	320 */
 /* #define __NR_signalfd	321 */
 /* #define __NR_timerfd_create	322 */
-/* #define __NR_eventfd		323 */
+#define __NR_eventfd		323
 #define __NR_fallocate		324
 #ifdef __KERNEL__
 
diff --git a/include/asm-x86_64/unistd.h b/include/asm-x86_64/unistd.h
index bb01083..63320e5 100644
--- a/include/asm-x86_64/unistd.h
+++ b/include/asm-x86_64/unistd.h
@@ -636,7 +636,7 @@ __SYSCALL(__NR_signalfd, sys_ni_syscall)
 #define __NR_timerfd_create	283
 __SYSCALL(__NR_timerfd_create, sys_ni_syscall)
 #define __NR_eventfd		284
-__SYSCALL(__NR_eventfd, sys_ni_syscall)
+__SYSCALL(__NR_eventfd, sys_eventfd)
 #define __NR_fallocate		285
 __SYSCALL(__NR_fallocate, sys_fallocate)
 
